File: java/src/Triangulo/TrianguloRetangulo_Main.java
package Triangulo;
import java.util.Scanner;
public class TrianguloRetangulo_Main {
    private static TrianguloRetangulo triRet;
    public static void main(String[] args) {
        //Alternativa 1 
        triRet = new TrianguloRetangulo();
        imprimeTriangulo(triRet);
        triRet = null;
        // Alternativa 2
        double[] lados = entraLados();
        triRet = new TrianguloRetangulo(lados[0], lados[1], lados[2]);
        imprimeTriangulo(triRet);
        triRet = null;
    }
    static void imprimeTriangulo(TrianguloRetangulo tr){
        if(tr == null){
            System.out.println(">>> ERRO: OBJETO NULO!");
            System.exit(0);
        }
        System.out.println("Valido? "+ tr.trianguloValido2());
        System.out.println("Lado1 = "+ tr.getLado1());
        System.out.println("Lado2 = "+ tr.getLado2());
        System.out.println("Lado3 = "+ tr.getLado3());
        System.out.println("Perimetro = " + tr.perimetro());
        System.out.println("Hipotenusa = "+ tr.hipotenusa());
        System.out.println("Area = "+ tr.area());
    }

    static double[] entraLados(){
        Scanner sc = new Scanner(System.in);
        double[] lados = new double[3];
        System.out.print("Digite Lado1 = ");
        lados[0] = sc.nextDouble();
        System.out.print("Digite Lado1 = ");
        lados[1] = sc.nextDouble();
        System.out.print("Digite Lado1 = ");
        lados[2] = sc.nextDouble();
        sc.close();
        return (lados);
    }
}

File: cpp/Triangulo/TrianguloRetanguloMain.cpp
#include <iostream>
#include <stdio.h>
#include "TrianguloRetangulo.h"

using namespace std;

float *entraLados();

void imprimeTriangulo(Triangulo *t);

TrianguloRetangulo *trisRet;

int main() {
    float *lados = entraLados();
    trisRet = new TrianguloRetangulo(lados[0], lados[1], lados[2]);
    imprimeTriangulo(trisRet);
    cout << "Triângulo válido: " << trisRet->trianguloValido2() << endl;
    double hipot = trisRet->hipotenusa();
    cout << "Hipotenusa: " << hipot << endl;
    double cateto = trisRet->cateto(hipot);
    cout << "Catetos: " << cateto << endl;
    cout << "Área: " << trisRet->area2() << endl;
    delete trisRet;

    cout << "\n\n>>> FIM DE EXECUÇÃO\n\n";
    return (0);
}

float *entraLados() {
    float *lados = new float[3];
    cout << "Lado 1 = ";
    cin >> lados[0];
    cout << "Lado 2 = ";
    cin >> lados[1];
    cout << "Lado 3 = ";
    cin >> lados[2];
    return lados;
}

void imprimeTriangulo(Triangulo *t) {
    if (t == NULL) {
        cout << "\n >>> ERRO: OBJETO NULO!";
        return;
    }
    cout << "\nLado1 = " << t->getLado1();
    cout << "\nLado2 = " << t->getLado2();
    cout << "\nLado3 = " << t->getLado3();
    cout << "\nPerimetro = " << t->perimetro();
    cout << "\nArea = " << t->area() << endl;
}